### ISSUE TITLE
[`metabase-lib`] Migrate `format_number` to cljs

### ISSUE THREAD

**Issue by camsaul** [metabase/metabase#27180]
***Migrate `format_number` to cljs***

Source:
- Implementation https://github.com/metabase/metabase/blob/master/frontend/src/metabase/lib/formatting/numbers.tsx#L61
- Unit tests https://github.com/metabase/metabase/blob/master/frontend/test/metabase/lib/formatting.unit.spec.js#L44

Signature:

```
format_number(value: number, options: NumberOptions): string

interface NumberOptions {
  compact?: boolean;
  currency?: string;
  currency_style?: "symbol" | "code" | "name";
  maximum_fraction_digits?: number;
  minimum_fraction_digits?: number;
  minimum_integer_digits?: number;
  maximum_significant_digits?: number;
  minimum_significant_digits?: number;
  negative_in_parentheses?: boolean;
  number_separators?: string;
  number_style?: "currency" | "decimal" | "scientific" | "percentage";
  scale?: number;
}
```

Changes to the existing FE function:
- Removed `jsx`. We would need to handle it on the FE https://github.com/metabase/metabase/blob/master/frontend/src/metabase/lib/formatting/numbers.tsx#L243
- Removed `decimals`. To get the same result set `minimum_fraction_digits` and `maximum_fraction_digits` to the same value.
- Removed both `type` and `currency_in_header`. It was used to remove the currency sign from cell values if `currency_in_header` is enabled. This is something very specific to tables and I believe it should be handled there and not in the core formatting function. Basically if `currency_in_header` is true than don't format the number as a `currency` in the table.

Tasks:
- [ ] Migrate format_number to cljs
- [ ] Use format_number from cljs in place of the existing formatting function

**Comment by camsaul** [comment]
@nemanjaglumac's notes from #26961 

> Just a reminder that we have another formatNumber utility in the static-viz code:
> https://github.com/metabase/metabase/blob/master/frontend/src/metabase/static-viz/lib/numbers.ts
> 
> and another set of unit tests related to it:
> https://github.com/metabase/metabase/blob/master/frontend/src/metabase/static-viz/lib/numbers.unit.spec.js

**Comment by camsaul** [comment]
Implemented by #27114
